### DISCUSSION TITLE
仓库改名

### DISCUSSION THREAD

**Discussion by aqni** [Tr0py/NKU-thesis-template-2020#28] [General]
续 https://github.com/Tr0py/NKU-thesis-template-2020/issues/27#issue-1710795169

github文档中关于仓库改名的说明

When you rename a repository, all existing information, with the exception of project site URLs, is automatically redirected to the new name, including: Issues, Wikis, Stars, Followers.
In addition to redirecting web traffic, all git clone, git fetch, or git push operations targeting the previous location will continue to function as if made on the new location.

但是文档中，对[GitHub Pages](https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#types-of-github-pages-sites).是否受到影响的说明语焉不详。

详见
https://docs.github.com/en/repositories/creating-and-managing-repositories/renaming-a-repository

**Comment by Tr0py** [comment]
看起来没什么问题，因为不会丢start和fork。唯一的影响是url的改变导致模板主页改变。当前是 https://tr0py.github.io/NKU-thesis-template-2020/ ，改了仓库名之后后面的部分会改变。不过问题不大，唯一用到此url链接的地方是overleaf模板。在overleaf模板更新之前得保证这个链接上有一个可以访问的网页。

我觉得临时的解决办法是，先改名，再创建一个同名repo，host github pages，这样新url和旧url都有github pages。

长期解决办法是，买个域名，把主页host在上面。比如`nkthesis.com`。

**Reply by Tr0py** [reply]
这么看改名应该可行，可以等refactor的todo都做完了改一下。

**Comment by Tr0py** [comment]
先记录下可以改的名字：我觉得`nkthesis-ug`是个不错的名字。

**Reply by aqni** [reply]
我在ctan上[搜索](https://www.ctan.org/search?phrase=thesis)了一下命名，模板一般使用xxx-thesis或者xxxthesis的形式。

国内其他模板的命名：
- [thuthesis](https://github.com/tuna/thuthesis)
- [fduthesis](https://github.com/stone-zeng/fduthesis)
- pkuthss
- [SJTUThesis](https://github.com/sjtug/SJTUThesis)

以上例子中，一般毕业论文通过选项选择本科生或硕博，一般用bachelor代指本科毕业论文。

我认为两种极端的选择是：
- `nkuthesis-bachelor` 最长
- `nkthss-ba` 最短

**Reply by aqni** [reply]
再给几个想法：
- `nkubachelor` 把thesis省略
- ~~`nkba` 简单粗暴，ba是bachelor的缩写~~ (NKBA好像经常被用作其他意思)
-

**Reply by aqni** [reply]
可以取一个有野心的名字：`nkuthesis`

**Reply by Tr0py** [reply]
> 可以取一个有野心的名字：`nkuthesis`

这个可以

**Reply by aqni** [reply]
因为计划中仓库中不仅有毕业论文模板，还有课程作业模板，所以也可以叫：`nkulatex`
